core/src/test/java/fr/sncf/osrd/envelope_sim_infra/MRSPTest.java

```diff
@@ -125,14 +127,82 @@ public void mrspFromInfraPath() {
         );
         var testRollingStock = TestTrains.VERY_SHORT_FAST_TRAIN;
-
-        var mrsp = MRSP.from(path, testRollingStock, true, null);
+        var driverBehaviour = new DriverBehaviour(0, 0);
+        var mrsp = MRSP.from(path, testRollingStock, true, null, driverBehaviour);
         Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(0, 1));
         Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(15, 1));
         Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(29, 1));
         Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(30, 1));
         Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(49, 1));
         Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(51, 1));
+        Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(56, 1));
+        Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(75, 1));
+    }
+
+    @Test
+    public void mrspWithDriverBehaviour() {
+        List<TrackRangeView> path = pathSenarioForTest();
+        var testRollingStock = TestTrains.VERY_SHORT_FAST_TRAIN;
+        var driverBehaviour = new DriverBehaviour(16, 17);
+        var mrsp = MRSP.from(path, testRollingStock, true, null, driverBehaviour);
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(0, 1));
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(12, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(13, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(29, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(30, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(33, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(35, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(47, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(56, 1));
+        Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(75, 1));
+    }
+
+    @Test
+    public void mrspWithNoDriverBehaviour() {
+        List<TrackRangeView> path = pathSenarioForTest();
+        var testRollingStock = TestTrains.VERY_SHORT_FAST_TRAIN;
+        var mrsp = MRSP.from(path, testRollingStock, true, null);
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(0, 1));
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(20, 1));
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(29, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(30, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(49, 1));
+        Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(56, 1));
         Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(75, 1));
```

Comment:
Isn't the speed supposed to be equal to 30 between 70 and 100 ?

Reply:
same test exists in mrsp